src/Framework/Sdk/SdkResult.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    ///     An abstract interface class to indicate SDK resolver success or failure.
+    /// </summary>
+    public abstract class SdkResult
+    {
+        /// <summary>
+        ///     Indicates the resolution was successful.
+        /// </summary>
+        public bool Success { get; protected set; }
```

Comment:
Since the SdkResultFactory mentions warnings and errors, maybe the SdkResult should do to. Either via Warning / Error properties, or a Diagnostic list, where the Diagnostic.Type is either Warning / Error.